### ISSUE TITLE
Setting a global task image

### ISSUE THREAD

**Issue by elliotgunton** [argoproj-labs/hera-workflows#331]
It would be great to be able to set a global image to be used as the default in the `Task.__init__` instead of the given default arg of `python3.7`, similar to the `get_global_api_version` change in https://github.com/argoproj-labs/hera-workflows/pull/327. It would help keep workflows using the same image throughout multiple tasks cleaner and readable.

The proposed changes would be:
* New `{get,set}_global_task_image` functions in `src/hera/host_config.py`
* Using the function in the `Task.__init__` default value for `image` [here](https://github.com/argoproj-labs/hera-workflows/blob/90e9275739639cc33e71b20668b679184b1d6d6f/src/hera/task.py#L183):
```py
image: str = get_global_task_image(),
```
* Test for `set/get_global_task_image`

Usage adapted on the `linear.py` example:
 ```py
"""This example showcases how to structure a linear chain of tasks"""
from hera import Task, Workflow, set_global_task_image

# assumes you used `hera.set_global_token` and `hera.set_global_host` so that the workflow can be submitted
with Workflow("linear") as w:
    set_global_task_image("my-host-images/python3.9")
    Task("t1", task_1) >> Task("t2", task_2) >> Task("t3", task_3) >> Task("t4", task_4)

w.create()
```

Alternatively if there's a way to keep this setting under a workflow that might be better and rename to `Workflow.set_default_task_image()`. But I don't think Tasks are currently explicitly coupled to the Workflow? Plus I don't think people would create multiple workflows in a single file that would require the image to be scoped e.g.
```py
with Workflow("linear") as w:
    w.set_default_task_image("my-host-images/python3.9")
    Task("t1", task_1) >> Task("t2", task_2) >> Task("t3", task_3) >> Task("t4", task_4)
```

Comments welcome! If this is something @flaviuvadan / @Trollgeir are happy with I can start a PR.

**Comment by flaviuvadan** [comment]
Thanks for the issue @elliotgunton! This sounds great! A PR will be most welcome :)

**Comment by flaviuvadan** [comment]
@elliotgunton we will release 4.1 once we merge your contribution so that your team gets to use the global image

**Comment by samj1912** [comment]
Just one minor comment, we should not use `get_global_task_image()` in the Task `__init__` function arguments. Let's use `None` and call the `get_global_task_image()` inside the `__init__` body.

So 

```
def __init__(self, image: Optional[str] = None, ...):
   self.image = image or get_global_task_image()
``` 

instead of 

```
def __init__(self, image: str = get_global_task_image(), ...):
   self.image = image
``` 

This is to avoid - https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

**Comment by samj1912** [comment]
Closed in #332
